components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -249,6 +267,175 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
     }
 }
 
+impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
+    pub fn on_ready_split_region<T>(
+        &mut self,
+        store_ctx: &mut StoreContext<EK, ER, T>,
+        derived_index: usize,
+        tablet_index: u64,
+        regions: Vec<Region>,
+    ) {
+        fail_point!("on_split", self.peer().get_store_id() == 3, |_| {});
+
+        let derived = &regions[derived_index];
+        let derived_epoch = derived.get_region_epoch().clone();
+        let region_id = derived.get_id();
+
+        // Group in-memory pessimistic locks in the original region into new regions.
+        // The locks of new regions will be put into the corresponding new regions
+        // later. And the locks belonging to the old region will stay in the original
+        // map.
+        let region_locks = {
+            let mut pessimistic_locks = self.txn_ext().pessimistic_locks.write();
+            info!(self.logger, "moving {} locks to new regions", pessimistic_locks.len(););
+            // Update the version so the concurrent reader will fail due to EpochNotMatch
+            // instead of PessimisticLockNotFound.
+            pessimistic_locks.version = derived_epoch.get_version();
+            pessimistic_locks.group_by_regions(&regions, derived)
+        };
+        fail_point!("on_split_invalidate_locks");
+
+        // Roughly estimate the size and keys for new regions.
+        let new_region_count = regions.len() as u64;
+        let mut meta = store_ctx.store_meta.lock().unwrap();
+        meta.set_region(
+            derived.clone(),
+            self,
+            RegionChangeReason::Split,
+            tablet_index,
+        );
+        self.post_split();
+
+        let last_region_id = regions.last().unwrap().get_id();
+        for (new_region, locks) in regions.into_iter().zip(region_locks) {
+            let new_region_id = new_region.get_id();
+            if new_region_id == region_id {
+                continue;
+            }
+
+            let split_init = PeerMsg::SplitInit(Box::new(SplitInit {
+                region: new_region,
+                parent_is_leader: self.is_leader(),
+                check_split: last_region_id == new_region_id,
+                locks,
+            }));
+
+            // First, send init msg to peer directly. Returning error means the peer is not
+            // existed in which case we should redirect it to the store.
+            match store_ctx.router.force_send(new_region_id, split_init) {
+                Ok(_) => {}
+                Err(SendError(PeerMsg::SplitInit(msg))) => {
+                    store_ctx.router.send_control(StoreMsg::SplitInit(msg));
+                }
+                _ => unreachable!(),
+            }
+        }
+        drop(meta);
+    }
+
+    pub fn on_split_init<T>(
```

Comment:
I didn't see where this is called.

Reply:
![image](https://user-images.githubusercontent.com/71589810/201263275-e098345d-494c-481c-969e-1cb35e698e03.png)

Reply:
Ah okay, I hide this file.